clang/test/Sema/compare.c

```diff
@@ -419,3 +419,27 @@ void pr36008(enum PR36008EnumTest lhs) {
   if (x == y) x = y; // no warning
   if (y == x) y = x; // no warning
 }
+
+int test13(unsigned a, int b) {
+        return a > ~(95 != b); // expected-warning {{comparison of integers of different signs}}
+}
+
+int test14(unsigned a, unsigned b) {
+        return a > ~b; // no-warning
+}
+
+int test15(unsigned a, int b) {
+        return a > -(95 != b); // expected-warning {{comparison of integers of different signs}}
+}
+
+int test16(unsigned a, unsigned b) {
+        return a > -b; // no-warning
+}
+
+int test17(int a, unsigned b) {
+    return a > --b; // expected-warning {{comparison of integers of different signs}}
```

Comment:
Is this supposed to be:
```suggestion
    return a > -(-b); // expected-warning {{comparison of integers of different signs}}
```
rather than a decrement?